pkg/api/server/register_images.go

```diff
@@ -709,18 +709,32 @@ func (s *APIServer) registerImagesHandlers(r *mux.Router) error {
 	//    description: username:password for the registry.
 	//    type: string
 	//  - in: query
+	//    name: os
+	//    description: Pull image for the specified operating system.
+	//    type: string
+	//  - in: query
+	//    name: arch
+	//    description: Pull image for the specified architecture.
+	//    type: string
+	//  - in: query
 	//    name: tls-verify
 	//    description: Require TLS verification.
 	//    type: bool
 	//    default: true
+	//  - in: query
+	//    name: all-tags
+	//    description: Pull all tagged images in the repository.
+	//    type: bool
 	// produces:
 	// - application/json
 	// responses:
 	//   200:
 	//     $ref: "#/response/LibpodImagesPullResponse"
+	//   400:
+	//     $ref: "#/responses/BadParamError"
 	//   500:
 	//     $ref: '#/responses/InternalError'
-	r.Handle(VersionedPath("/libpod/images/pull"), APIHandler(s.Context, libpod.ImagesPull)).Methods(http.MethodPost)
+	r.Handle(VersionedPath("/libpod/images/pull"), APIHandler(s.Context, libpod.ImagesPull)).Methods(http.MethodGet)
```

Comment:
This operation is Post not Get.

Reply:
Can you elaborate why? I changed it to Get because everything is done via parameters.

Reply:
I peeked into https://www.w3schools.com/tags/ref_httpmethods.asp and agree that it should be POST.